helm/promtail/templates/service.yaml

```diff
@@ -19,4 +20,5 @@ spec:
   ports:
     - protocol: TCP
       port: 9091
-      targetPort: {{ .Values.config.serverPort }}
+      targetPort: {{ .Values.promtail.config.serverPort }}
```

Comment:
If you use the port's friendly name (I guess `http-metrics`) you don't have to rely on values.
On top of that, that avoids values visibility issues.

Reply:
Changed targetPort with hardcoded value